cosi/cosi_new.go

```diff
@@ -0,0 +1,130 @@
+package cosi
+
+import (
+	"crypto/cipher"
+
+	"github.com/dedis/kyber/abstract"
+)
+
+func Commit(suite abstract.Suite, s cipher.Stream) (abstract.Scalar, abstract.Point) {
+	// compute random scalar
+	// compute commitment
+	// return both
+}
+
+func AggregateCommitments(suite abstract.Suite, commitments []abstract.Point, mask *Mask) (abstract.Point, *Mask) {
+	// sum up all commitments
+	// update mask
+	// return both
+}
+
+func Challenge(suite abstract.Suite, commitment abstract.Point, pubkey abstract.Point, mask, message []byte) abstract.Scalar {
+	// return H(commitment || pubkey || mask || message)
+}
+
+func Response(suite abstract.Suite, random abstract.Scalar, challenge abstract.Scalar, prikey abstract.Scalar) {
+	// return random - challenge * prikey
+}
+
+func AggregateResponses(suite abstract.Suite, responses []abstract.Scalar) abstract.Scalar {
```

Comment:
As far as I know we want to be able to aggregate responses from our children and then pass up the result to our parent in the tree communication model. So we cannot integrate response aggregation directly?